components/Repositories/LotteryFactorChart.tsx

```diff
@@ -0,0 +1,210 @@
+      case "very-high":
+        return "#E80010";
+      case "high":
+        return "#FC5108";
+      case "moderate":
+        return "#FEC009";
+      case "low":
+        return "#5CC90C";
+      case "Other Contributors":
+        return "#0649F1";
+    }
+  }
+
+  return (
+    <Card className="flex flex-col gap-4 w-full h-full items-center pt-8 p-4">
+      <section className="flex flex-col lg:flex-row w-full items-start lg:items-center gap-4 lg:justify-between px-4 lg:px-8">
+        <header className="flex w-full justify-between items-center">
+          <div className="flex gap-2 items-center">
+            <HiOutlineTicket className="text-xl" />
+            <h3 className="text-sm font-semibold md:text-lg text-slate-800">Lottery Factor</h3>
+          </div>
+          {!lotteryFactor ? (
+            <Pill text="..." color="slate" />
+          ) : (
+            <Pill
+              text={lotteryFactor.all_lotto_factor.replace("-", " ") ?? ""}
+              color={
+                lotteryFactor.all_lotto_factor === "high"
+                  ? "red"
+                  : lotteryFactor.all_lotto_factor === "moderate"
+                  ? "yellow"
+                  : lotteryFactor.all_lotto_factor === "low"
+                  ? "green"
+                  : lotteryFactor.all_lotto_factor === "very-high"
+                  ? "purple"
+                  : "slate"
+              }
+              className="capitalize"
+            />
+          )}
+        </header>
+      </section>
+
+      <section className="w-full px-4 lg:px-8 flex flex-col gap-4 text-sm lg:text-base">
+        {summary}
+        <div className="flex gap-1 h-3 place-content-center">
+          {sortedContributors.map((item, index) => {
+            return (
+              <button
+                aria-label={`${item.name} is ${item.value}% of the most used languages for contributors in your list`}
+                key={item.name}
+                data-language={item.name}
+                className={`${index === 0 ? "rounded-l-lg" : ""} ${
+                  index === sortedContributors.length - 1 ? "rounded-r-lg" : ""
+                } transform hover:scale-110 transition-transform hover:z-10`}
+                style={{ backgroundColor: getLottoColor(item.factor), width: `${item.value}%` }}
+                onMouseOver={(event) => {
+                  const { language } = event.currentTarget.dataset;
+                  setHovered(language);
+                }}
+                onMouseOut={(event) => {
+                  setHovered(undefined);
+                }}
+                onFocus={(event) => {
+                  const { language } = event.currentTarget.dataset;
+                  setHovered(language);
+                }}
+                onBlur={(event) => {
+                  setHovered(undefined);
+                }}
+              />
+            );
+          })}
+        </div>
+      </section>
+
+      {!lotteryFactor ? (
+        <p>Loading...</p>
```

Comment:
Do we want a better loading state @isabensusan?

Reply:
@zeucapua @nickytonline 
can we do something like this?

- skeleton loader component in the top right corner badge
- skeleton loader component in the header, for the text and the bar chart
- skeleton loader component inside each row in the table

![image](https://github.com/open-sauced/app/assets/27322879/ec4ea6d6-f8f1-4d22-ac27-80a7b14bcd86)